crates/rpc/rpc/src/debug.rs

```diff
@@ -534,6 +537,138 @@ where
             .await
     }
 
+    /// The `debug_executionWitness` method allows for re-execution of a block with the purpose of
+    /// generating an execution witness. The witness comprises of a map of all hashed trie nodes
+    /// to their preimages that were required during the execution of the block, including during
+    /// state root recomputation.
+    pub async fn debug_execution_witness(
+        &self,
+        block: BlockNumberOrTag,
+    ) -> EthResult<HashMap<B256, Bytes>> {
+        let block = match self.inner.eth_api.block(block.into()).await? {
+            None => return Err(EthApiError::UnknownBlockNumber),
+            Some(res) => res,
+        };
+        let (cfg, block_env, _) = self.inner.eth_api.evm_env_at(block.hash().into()).await?;
+        let this = self.clone();
+
+        self.inner
+            .eth_api
+            .spawn_with_state_at_block(block.parent_hash.into(), move |state| {
+                let mut db = StateBuilder::new()
+                    .with_database(StateProviderDatabase::new(state))
+                    .with_bundle_update()
+                    .build();
+
```

Comment:
reminder to self: Need 4788 call here

Reply:
wondering if we can eventually use the executor traits here, as there will be more system contract calls and maintaining a copy of block execution logic in debug APIs seems brittle

Reply:
Agree. I also specifically need this for Optimism execution, so that would be ideal. Glad to circle back around to this before we merge, just have to get it working first.